exp/internal/controllers/machinepool_controller_noderef.go

```diff
@@ -68,7 +68,7 @@ func (r *MachinePoolReconciler) reconcileNodeRefs(ctx context.Context, cluster *
 	log = log.WithValues("cluster", cluster.Name)
 
 	// Check that the MachinePool has valid ProviderIDList.
-	if len(mp.Spec.ProviderIDList) == 0 {
+	if len(mp.Spec.ProviderIDList) == 0 && (mp.Spec.Replicas == nil || *mp.Spec.Replicas != 0) {
```

Comment:
I'm not sure I fully understand `&& (mp.Spec.Replicas == nil || *mp.Spec.Replicas != 0)`.
is the intent to log the message in all the cases except when replicas are zero?

Reply:
yeah exactly. When ProviderIDList is non-empty and replicas > 0, this will log. That's probably still interesting information because if that stays for a while something is off with the ProviderIDList of the infra provider. 

Before the change, when scaling to zero this would log always, so I had to adjust it.

Reply:
Thanks for clarifying